src/test/scala/mesosphere/marathon/state/GroupManagerTest.scala

```diff
@@ -57,17 +57,40 @@ class GroupManagerTest extends TestKit(ActorSystem("System")) with MockitoSugar
     val group = Group(PathId.empty, Set(
       AppDefinition("/app1".toPath, ports = Seq(), container = Some(container))
     ))
-    val update = manager(10, 20).assignDynamicAppPort(Group.empty, group)
+    val update = manager(minServicePort = 10, maxServicePort = 20).assignDynamicServicePorts(Group.empty, group)
     update.transitiveApps.filter(_.hasDynamicPort) should be ('empty)
     update.transitiveApps.flatMap(_.ports.filter(x => x >= 10 && x <= 20)) should have size 2
   }
 
+  // Regression test for #1365
```

Comment:
Could you add tests for: 
-only ports (no container)
-mixed port and container